File: CS/Acacia Builder/Ttyn/GameStates/GameStateManager.cs
using Audrey;
using Microsoft.Xna.Framework;
using Microsoft.Xna.Framework.Graphics;
using Microsoft.Xna.Framework.Input;
using Tyn;

public class GameStateManager {

    public GameState gameState;
    private GameState menuState;
    public GameState playState;
    public GameState battleState;
    Game1 game;

    public enum State {
        MENU,
        PLAY,
        BATTLE,
        GAMEOVER
    }

    public GameStateManager(Game1 game) {
        this.game = game;
        menuState = new MenuState(game, this);
        playState = new PlayState(game, this);
        setState(State.MENU);
    }

    public void setState(State state){

        switch (state) {
            case State.MENU:
                gameState = menuState;
                break;
            case State.PLAY:
                gameState = playState;
                break;
            case State.BATTLE:
                gameState = battleState;
                break;
        }
    }
    public void update(GameTime gametime){
        gameState.update(gametime);
    }
    public void draw(){
        gameState.draw();
    }
}

File: Java/Astroids/core/src/com/neet/managers/GameStateManager.java
package com.neet.managers;

import com.neet.gamestates.*;

public class GameStateManager {
    //current game state
    private GameState gameState;

    public static final int MENU = 0;
    public static final int PLAY = 893746;
    public static final int HIGHSCORE = 234;
    public static final int GAMEOVER = 1234;

    public GameStateManager(){
        setState(MENU);
    }

    public void setState(int state){

        if (gameState != null) gameState.dispose();
        if(state == MENU){
            gameState = new MenuState(this);
        }
        if(state == PLAY){
            gameState = new PlayState(this);
        }
        if(state == HIGHSCORE){
            gameState = new HighScoreState(this);
        }
        if(state == GAMEOVER){
            gameState = new GameOverState(this);
        }
    }
    public void update(float dt){
        gameState.update(dt);
    }
    public void draw(){
        gameState.draw();
    }

}
